### PR TITLE
fixed markov file loading and saving

### PR DESCRIPTION
**Pull request by piotrowskv**

### CODE REVIEW

**Review by shilangyu — APPROVED** [review]
lgtm

**Comment by shilangyu** [comment]
Hm tests for saving/loading are failing

File: models/models/markov_chain/markov_chain.py [hunk 1]
```diff
@@ -167,7 +170,8 @@ def predict(self, initial_notes: tuple, length: int, deterministic: bool, rand:
                     list(probs.keys()), weights=probs.values(), k=1)[0]
 
             previous_n_gram = previous_n_gram[1:] + (next_note,)
-            prediction.append(next_note)
+            if next_note is not None:
+                prediction.append(next_note)
```
Comment (szgorski) [inline models/models/markov_chain/markov_chain.py:175]:
```
Run pipenv run mypy models/ --disallow-untyped-defs --ignore-missing-imports
models/markov_chain/markov_chain.py:174: error: Argument 1 to "append" of "list" has incompatible type "Tuple[Any, ...]"; expected "Tuple[Any]"  [arg-type]

Error: Process completed with exit code 1.
```

File: models/tests/test_markov_chain.py [hunk 2]
```diff
@@ -56,7 +57,11 @@ def test_multiple_tracks():
 
 
 def test_model_saving_and_loading(tmpdir):
-    path = Path(tmpdir).joinpath('markov_saved.npy')
+
+    if not os.path.exists(tmpdir):
+        os.makedirs(tmpdir, exist_ok=True)
+
+    path = Path(tmpdir).joinpath('markov_saved.npz')
```
Comment (szgorski) [inline models/tests/test_markov_chain.py:64]: `.npz` or `.npy`?

File: models/models/markov_chain/markov_chain.py (hunk 1 above)
Comment (szgorski) [reply models/models/markov_chain/markov_chain.py:175]: typing must be updated accordingly

**Review by szgorski — CHANGES_REQUESTED** [review]

File: models/tests/test_markov_chain.py (hunk 2 above)
Comment (shilangyu) [reply models/tests/test_markov_chain.py:64]: Npz is correct, it is a collection of `npy`s

Comment (szgorski) [reply models/tests/test_markov_chain.py:64]: kk, thanks for the insight :>

**Pull request closed without merging**